internal/crossplane/service_mariadb_database.go

```diff
@@ -134,11 +142,29 @@ func (msb MariadbDatabaseServiceBinder) GetBinding(ctx context.Context, bindingI
 	}
 
 	pw := string(secret.Data[xrv1.ResourceCredentialsSecretPasswordKey])
-	creds := createCredentials(endpoint, bindingID, pw, msb.instance.Composite.GetName())
+	creds := createCredentials(endpoint, bindingID, pw, msb.id)
 
 	return creds, nil
 }
 
+func (msb MariadbDatabaseServiceBinder) ValidateProvisionParams(ctx context.Context, params json.RawMessage) (map[string]interface{}, error) {
+	p := MariaDBProvisionAdditionalParams{}
+	if err := json.Unmarshal(params, &p); err != nil {
+		return nil, err
+	}
+	if p.ParentReference == "" {
+		return nil, errors.New("parent_reference required")
+	}
+
+	cmp := composite.New(composite.WithGroupVersionKind(mariaDBGroupVersionKind))
+	if err := msb.cp.client.Get(ctx, types.NamespacedName{Name: p.ParentReference}, cmp); err != nil {
+		return nil, fmt.Errorf("valid parent_reference required: %w", err)
+	}
+	return map[string]interface{}{
```

Comment:
couldn't you just marshal `MariaDBProvisionAdditionalParams` here? That way we don't have to use the `parent_reference` string again.

Reply:
for that I'd need to use e.g. https://github.com/mitchellh/mapstructure to convert the struct to a map again. 

I'd rather use the constant instead, I already had that change but I think it got lost in some `git stash`-ing ;)